### PR TITLE
Make FBPs Constructable

### PR DESCRIPTION
**Pull request by Arokha**
Print 'em out, slap an MMI or Posibrain in. They come from a new machine, the Prosthetics Fabricator. You can order license discs from cargo to print the proprietary limbs. Making FBPs requires a bit of research as well, otherwise you only get arms/legs. Also requires a bit of plasteel to make the critical parts. Mostly just need every limb, plus eyes. Torso comes with the snowflakey FBP battery.

Also adds an FBP 'self-diagnostics' verb. More vague than an analyzer.

:warning: **This adds a machine that needs to be placed on the map. "/obj/machinery/pros_fabricator" Not adding this machine will prevent you from constructing cyborgs.**

Personally, I would replace the right-side exosuit fab with it. I can add this to my PR but I'm unsure how you guys feel about non-devs submitting map changes.

### CODE REVIEW

**Comment by Arokha** [comment]
Hmm. That's just how that works, Travis. I'll have to snowflake it to fix that one.

File: code/game/mecha/mech_prosthetics.dm [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+		if(/obj/item/stack/material/phoron)
+			material = "phoron"
+		if(/obj/item/stack/material/steel)
+			material = DEFAULT_WALL_MATERIAL
+		if(/obj/item/stack/material/glass)
+			material = "glass"
+		if(/obj/item/stack/material/uranium)
+			material = "uranium"
+		if(/obj/item/stack/material/plasteel)
+			material = "plasteel"
```
Comment (GinjaNinja32) [inline code/game/mecha/mech_prosthetics.dm]: maaaaateeeerial systeeeeeeem

use the vars on the stack to get its material directly, not a huge switch :o

File: code/game/mecha/mech_prosthetics.dm [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+		if("silver")
+			mattype = /obj/item/stack/material/silver
+		if("diamond")
+			mattype = /obj/item/stack/material/diamond
+		if("phoron")
+			mattype = /obj/item/stack/material/phoron
+		if("uranium")
+			mattype = /obj/item/stack/material/uranium
+		if("plasteel")
+			mattype = /obj/item/stack/material/plasteel
```
Comment (GinjaNinja32) [inline code/game/mecha/mech_prosthetics.dm]: ditto, get the material from material system and tell it to place sheets :o

Comment (Arokha) [inline code/game/mecha/mech_prosthetics.dm]: It's copypasta'd from the exosuit fab. I'm unfamiliar with the material system, regardless.

Comment (GinjaNinja32) [inline code/game/mecha/mech_prosthetics.dm]: https://github.com/PolarisSS13/Polaris/blob/master/code/modules/materials/materials.dm

Stacks have a `material` var holding their material; if you *need* a string, `material.name` is the material names you have here. `get_material_by_name(name)` will get you the material datum from the string; you'll want `[material].place_sheet(sheet_loc)`.

**Comment by Arokha** [comment]
So um, did you guys want this, or like, is there something else I could add that you were waiting on? Someone mentioned you didn't like the cargo purchasing mechanic? Do you want me to remove that? Orrr like, I could do the map change for you, or...

File: code/game/mecha/mech_fabricator.dm [hunk 3]
```diff
@@ -145,43 +145,32 @@
...
-			material = "phoron"
-		if(/obj/item/stack/material/steel)
-			material = DEFAULT_WALL_MATERIAL
-		if(/obj/item/stack/material/glass)
-			material = "glass"
-		if(/obj/item/stack/material/uranium)
-			material = "uranium"
+	if(istype(I,/obj/item/stack/material))
+		var/obj/item/stack/material/S = I
+		if(!S.material in materials)
```
Comment (GinjaNinja32) [inline code/game/mecha/mech_fabricator.dm]: `if(!(S.material in materials))`

File: code/game/mecha/mech_prosthetics.dm [hunk 4]
```diff
@@ -172,46 +172,32 @@
...
-			material = DEFAULT_WALL_MATERIAL
-		if(/obj/item/stack/material/glass)
-			material = "glass"
-		if(/obj/item/stack/material/uranium)
-			material = "uranium"
-		if(/obj/item/stack/material/plasteel)
-			material = "plasteel"
+	if(istype(I,/obj/item/stack/material))
+		var/obj/item/stack/material/S = I
+		if(!S.material in materials)
```
Comment (GinjaNinja32) [inline code/game/mecha/mech_prosthetics.dm]: Ditto. Why is this code copy-pasted? Make a type of machine that accepts material sheets and base these two machines on that.

Comment (GinjaNinja32) [inline code/game/mecha/mech_prosthetics.dm]: This says it's outdated because the line below it changed; it is still relevant.

Comment (Arokha) [inline code/game/mecha/mech_prosthetics.dm]: It's sort of outside the scope of this project to rewrite how other machines work, to be honest. Your not-this catch is good though so I'll change that real quick.

**Comment by Searif** [comment]
I like you

File: code/modules/research/designs.dm [hunk 5]
```diff
@@ -206,60 +206,6 @@ other types of metals and chemistry for reagents).
...
-
-/datum/design/item/powercell/hyper
-	name = "hyper-capacity"
-	id = "hyper_cell"
-	req_tech = list(TECH_POWER = 5, TECH_MATERIAL = 4)
-	materials = list(DEFAULT_WALL_MATERIAL = 400, "gold" = 150, "silver" = 150, "glass" = 70)
-	build_path = /obj/item/weapon/cell/hyper
-	category = "Misc"
-	sort_string = "DAAAD"
-
```
Comment (EmperorJon) [inline code/modules/research/designs.dm]: Looks like this went through and was missed somewhat, as it seems people weren't expecting it.

Comment (Yoshax) [inline code/modules/research/designs.dm]: It got fixed 7 hours ago. It wasn't deliberate, I believe.

**Comment by Searif** [comment]
I know I said I like you, but I have an issue with only unbranded being available by default, grumble grumble.

**Pull request merged**